src/app.ts

```diff
@@ -0,0 +1,77 @@
+import express from 'express';
+import nunjucks from 'nunjucks';
+import { config } from './config/index.js';
+import { errorHandler, notFoundHandler, requestLogger } from './middleware/index.js';
+import routes from './routes/index.js';
+
+/**
+ * Create and configure the Express application
+ */
+export const createApp = (): express.Application => {
+  const app = express();
+
+  // Configure Nunjucks template engine
+  nunjucks.configure(config.paths.views, {
+    autoescape: config.template.autoescape,
+    express: app,
+    watch: config.template.watch,
+  });
+
+  // Set template engine
+  app.set('view engine', config.template.engine);
+  app.set('views', config.paths.views);
+
+  // Middleware setup
+  app.use(requestLogger);
+  app.use(express.json());
+  app.use(express.urlencoded({ extended: true }));
+
+  // Serve static files
+  app.use(express.static(config.paths.public));
+
+  // Routes
+  app.use('/', routes);
+
+  // Error handling
+  app.use(notFoundHandler);
+  app.use(errorHandler);
```

Comment:
Maybe not a point for now, but having a generic error handler and then decide what to do based on the HTTP Error code could be good for the future